perllib/FixMyStreet/Cobrand/UKCouncils.pm

```diff
@@ -171,17 +171,88 @@ sub area_check {
 }
 
+=head2 responsible_for_areas
+
+Tests to see if the cobrand is responsible for a location. This is done through
+checking if it has the area set in council_area_id and also checking if the
+coverage is limited only to an overlapping asset by testing the hook
+report_is_on_cobrand_asset
+
+=cut
+
+sub responsible_for_areas {
+    my ($self, $councils) = @_;
+
+    if ($self->{c}->cobrand->can('report_is_on_cobrand_asset')) {
+        if (scalar(%$councils) == 1 && (values %$councils)[0]->{name} eq $self->council_name) {
+            return 1;
+        } elsif ($self->{c}->cobrand->call_hook('report_is_on_cobrand_asset')) {
+            return 1;
+        } else {
+            return 0;
+        }
+    } else {
+        # The majority of cobrands only cover a single area, but e.g. Northamptonshire
+        # covers multiple so we need to handle that situation.
+        my $council_area_ids = $self->council_area_id;
+        $council_area_ids = [ $council_area_ids ] unless ref $council_area_ids eq 'ARRAY';
+        foreach (@$council_area_ids) {
+            return 1 if defined $councils->{$_};
+        }
+
+        return 0;
+    }
+}
+
+=head2 munge_overlapping_asset_bodies
+
+Some bodies have assets which overlap, or are in, another body's area.
+For those, we add the overlapped area into the council_area_id so the point
+in the other area can be selected. But we then need to filter which categories
+appear - if it's on the overlapping asset, the cobrand body, otherwise
+whichever body actually covers that area.
+
+=cut
+
+sub munge_overlapping_asset_bodies {
+    my ($self, $bodies) = @_;
+
+    return unless $self->can('report_is_on_cobrand_asset');
+
+    my %this_body = map { $_->id => $_ } grep { $_->name eq $self->council_name || $_->name eq $self->body->name } values %$bodies;
+    my %other_bodies = map { $_->id => $_ } grep { $_->name ne $self->council_name && $_->name ne $self->body->name } values %$bodies;
+
+    if (scalar(values %$bodies) == 1 && (grep { $self->council_name eq $_->name } values %$bodies || grep { $self->body->name eq $_->name } values %$bodies)) {
+        # The cobrand body is the only body, so do nothing.
+        return;
+    }
+
+    if (scalar(values %$bodies) == 2 && grep { $_->name =~ /TfL|National Highways/} values %$bodies) {
+        #The cobrand body and National Highways or TfL are the only bodies, so do nothing.
+        return;
+    }
+
+    if (scalar(values %$bodies) == 3 && (grep { $_->name eq 'National Highways' } values %$bodies
+        && grep { $_->name eq 'TfL' } values %$bodies)) {
+        #The cobrand body and National Highways and TfL are the only bodies, so do nothing.
+        return;
+    }
+
+    # If we get here, assume we are outside the cobrand's boundaries so if there's a test
+    # we run it to see if we are on the cobrand's asset and return the relevant bodies
+    if ($self->call_hook('report_is_on_cobrand_asset' => \%this_body, \%other_bodies)) {
```

Comment:
How can the hook take parameters here, but not when called in responsible_for_areas?
You're calling the function twice here as well, either check+call direct, or use call_hook.